pandas/tests/groupby/test_apply.py

```diff
@@ -543,6 +543,26 @@ def predictions(tool):
     tm.assert_series_equal(expected, result)
 
 
+def test_apply_aggregating_timedelta_and_datetime():
+    # Regression test for GH 15562
+    # The following groupby caused ValueErrors and IndexErrors pre 0.20.0
+
+    s = """clientid,datetime
+    A,2017-02-01 00:00:00
+    B,2017-02-01 00:00:00
+    C,2017-02-01 00:00:00"""
+
+    df = pd.read_csv(StringIO(s))
+    df["datetime"] = pd.to_datetime(df.datetime)
+    df["time_delta_zero"] = df.datetime - df.datetime
+    result = df.groupby("clientid").apply(
+        lambda ddf: pd.Series(
+            dict(clientid_age=ddf.time_delta_zero.min(), date=ddf.datetime.min())
+        )
+    )
+    assert result.shape == (3, 2)
```

Comment:
Can you construct the expected dataframe and use `tm.assert_frame_equal`?

Reply:
yes. just pushed the update.